src/cascadia/TerminalApp/VsSetupConfiguration.h

```diff
@@ -0,0 +1,142 @@
+Licensed under the MIT license.
+
+Module Name:
+- VsSetupConfiguration
+
+Abstract:
+- Encapsulates the Visual Studio Setup Configuration COM APIs
+
+Author(s):
+- Charles Willis - October 2020
+
+--*/
+
+#pragma once
+
+#include <wil/resource.h>
+#include "Setup.Configuration.h"
+
+namespace TerminalApp
+{
+    /// <summary>
+    /// See https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.setup.configuration?view=visualstudiosdk-2019
+    /// </summary>
+    class VsSetupConfiguration
+    {
+    private:
+        typedef wil::com_ptr<ISetupConfiguration2> ComPtrSetupQuery;
+        typedef wil::com_ptr<ISetupHelper> ComPtrSetupHelper;
+        typedef wil::com_ptr<ISetupInstance> ComPtrSetupInstance;
+        typedef wil::com_ptr<ISetupPropertyStore> ComPtrPropertyStore;
+        typedef wil::com_ptr<ISetupPackageReference> ComPtrPackageReference;
+        typedef wil::com_ptr<ISetupInstanceCatalog> ComPtrInstanceCatalog;
+
+    public:
+        struct VsSetupInstance
+        {
+            inline const static std::wstring_view kVersion16_3{ L"16.3.0.0" };
+
+            inline std::wstring ResolvePath(std::wstring_view relativePath) const
+            {
+                return VsSetupConfiguration::ResolvePath(mp_inst, relativePath);
+            }
+
+            inline std::wstring GetDevShellModulePath() const
+            {
+                unsigned long long ullInstanceVersion{ 0 };
+                THROW_IF_FAILED(mp_helper->ParseVersion(GetVersion().data(), &ullInstanceVersion));
+
+                unsigned long long ullVersion16_3{ 0 };
+                THROW_IF_FAILED(mp_helper->ParseVersion(kVersion16_3.data(), &ullVersion16_3));
+
+                if (ullInstanceVersion >= ullVersion16_3)
+                    return ResolvePath(L"Common7\\Tools\\Microsoft.VisualStudio.DevShell.dll");
+
+                return ResolvePath(L"Common7\\Tools\\vsdevshell\\Microsoft.VisualStudio.DevShell.dll");
+            }
+
+            inline std::wstring GetDevCmdScriptPath() const
+            {
+                return ResolvePath(L"Common7\\Tools\\VsDevCmd.bat");
+            }
+
+            inline bool VersionInRange(std::wstring_view range) const
+            {
+                return VsSetupConfiguration::InstallationVersionInRange(mp_query, mp_inst, range);
+            }
+
+            inline std::wstring GetVersion() const
+            {
+                return VsSetupConfiguration::GetInstallationVersion(mp_inst);
+            }
+
+            inline std::wstring GetInstallationPath() const
+            {
+                return VsSetupConfiguration::GetInstallationPath(mp_inst);
+            }
+
+            inline std::wstring GetInstanceId() const
+            {
+                return VsSetupConfiguration::GetInstanceId(mp_inst);
+            }
+
+            inline std::wstring GetChannelId() const
+            {
+                ComPtrPropertyStore properties = mp_inst.query<ISetupPropertyStore>();

```

Comment:
Older versions of the query API didn't implement `ISetupPropertyStore`. I [added that in 15.3](https://devblogs.microsoft.com/setup/new-vswhere-with-more-properties/), so any customers on VS2017 RTM servicing - or with versions older than 15.3 otherwise - would throw here.

Reply:
In the case where this is not defined, just elide the channel from the display name i.e. return an empty string.